### PR TITLE
Feature/adc/#224 genre view

### PR DESCRIPTION
**Pull request by AmauryDanisCousandier**
- genre detail view
- artist detail view

### CODE REVIEW

File: front/API.ts [hunk 1]
```diff
@@ -488,6 +542,16 @@ export default class API {
 		};
 	}
 
+	// public static getFavorites(): Query<Song[]> {
```
Comment (Arthi-chaud) [inline front/API.ts:545]: Pleqse, remove this.

File: front/Theme.tsx [hunk 2]
```diff
@@ -4,9 +4,15 @@ import { useEffect } from 'react';
 
 const ThemeProvider = ({ children }: { children: JSX.Element }) => {
 	const colorScheme = useColorScheme();
+    const config = {
+        dependencies: {
+            "linear-gradient": require("expo-linear-gradient").LinearGradient,
```
Comment (Arthi-chaud) [inline front/Theme.tsx:9]: Please, use a proper import statement. Just tested it, it works :]

File: front/components/RowCustom.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { useColorScheme } from 'react-native';
+import { RootState, useSelector } from '../state/Store';
+import { Box, Pressable } from 'native-base';
+
+const RowCustom = (props: Parameters<typeof Box>[0] & { onPress?: () => void }) => {
```
Comment (Arthi-chaud) [inline front/components/RowCustom.tsx:5]: What is this component supposed to do? Isn't there a Row component already?

File: front/components/SearchBar.tsx [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { translate } from '../i18n/i18n';
 import { SearchContext } from '../views/SearchView';
 import { debounce } from 'lodash';
 
-export type Filter = 'artist' | 'song' | 'genre' | 'all';
+export type Filter = 'artist' | 'song' | 'genre' | 'all' | 'favorite';
```
Comment (Arthi-chaud) [inline front/components/SearchBar.tsx:8]: Favorites are not part of this PR, are they? Please, remove this for now

File: front/components/SearchResult.tsx [hunk 5]
```diff
@@ -351,7 +258,8 @@ const FilterSwitch = () => {
 
 export const SearchResultComponent = () => {
 	const { stringQuery } = React.useContext(SearchContext);
-	const shouldOutput = !!stringQuery.trim();
+	const { filter } = React.useContext(SearchContext);
+	const shouldOutput = !!stringQuery.trim() || filter == "favorite";
```
Comment (Arthi-chaud) [inline front/components/SearchResult.tsx:262]: Same as before, remove this while it is not relevant

File: front/components/SongRow.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import { MaterialIcons } from "@expo/vector-icons";
+
+type SongRowProps = {
+	song: Song | SongWithArtist; // TODO: remove Song
+	onPress: () => void;
+};
+
+const SongRow = ({ song, onPress }: SongRowProps) => {
+	return (
+		<RowCustom width={'100%'}>
```
Comment (Arthi-chaud) [inline front/components/SongRow.tsx:14]: This entier component has already been done. Please see `SongRow`

File: front/views/ArtistDetailsView.tsx [hunk 7]
```diff
@@ -1,49 +1,61 @@
...
-					/>
-				</VStack>
-			</Container>
-		</SafeAreaView>
+		<ScrollView>
+			<ImageBackground
+				style={{width : '100%', height: isMobileView ? 200 : 300}}
+				source={{uri : "https://picsum.photos/720"}}>
+				<LinearGradient 
+					colors={['#00000000', '#000000']}
```
Comment (Arthi-chaud) [inline front/views/ArtistDetailsView.tsx:37]: Where are these values coming from?

File: front/views/ArtistDetailsView.tsx [hunk 8]
```diff
@@ -1,49 +1,61 @@
...
-						variant="unstyled"
-						_pressed={{ opacity: 0.6 }}
-					/>
-				</VStack>
-			</Container>
-		</SafeAreaView>
+		<ScrollView>
+			<ImageBackground
+				style={{width : '100%', height: isMobileView ? 200 : 300}}
+				source={{uri : "https://picsum.photos/720"}}>
```
Comment (Arthi-chaud) [inline front/views/ArtistDetailsView.tsx:35]: Why not use the image from the API?

File: front/views/GenreDetailsView.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
+import { Box, Flex, Heading, useBreakpointValue, ScrollView } from 'native-base';
+import { useQueries, useQuery } from '../Queries';
+import { LoadingView } from '../components/Loading';
+import { RouteProps, useNavigation } from '../Navigation';
+import API from '../API';
+import CardGridCustom from '../components/CardGridCustom';
+import SongCard from '../components/SongCard';
+
+const colorRange = ['#364fc7', '#5c940d', '#c92a2a', '#d6336c', '#20c997'];
```
Comment (Arthi-chaud) [inline front/views/GenreDetailsView.tsx:9]: Where are these values coming from?

File: front/views/GenreDetailsView.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+				},}}
+			/>
+			<Heading ml={3} fontSize={50}>{genreQuery.data.name}</Heading>
+			<Flex
+				flexWrap="wrap"
+				direction={isMobileView ? 'column' : 'row'}
+				justifyContent={['flex-start']}
+				mt={4}
+			>
+				<CardGridCustom
```
Comment (Arthi-chaud) [inline front/views/GenreDetailsView.tsx:50]: Use `SongCardGrid` instead

File: front/views/ArtistDetailsView.tsx [hunk 11]
```diff
@@ -1,49 +1,61 @@
...
+			</ImageBackground>
+			<Box>
+				<Heading mt={-20} ml={3} fontSize={50}>{artistQuery.data.name}</Heading>
+				<ScrollView mt={3}>
+					<Box>
+						{songsQuery.data.map((comp: Song, index: Key | null | undefined) => (
+							<SongRow
+								key={index}
+								song={comp}
+								liked={true}
```
Comment (Arthi-chaud) [inline front/views/ArtistDetailsView.tsx:48]: This property is superfluous and keeps the code from type-checking

File: front/Theme.tsx (hunk 2 above)
Comment (AmauryDanisCousandier) [reply front/Theme.tsx:9]: wdym ?

File: front/components/RowCustom.tsx (hunk 3 above)
Comment (AmauryDanisCousandier) [reply front/components/RowCustom.tsx:5]: RowCustom is used to create other rows such as the SongRow, it's there so you don't repete the code always used to make rows, thought it would be smarter

File: front/components/SearchBar.tsx [hunk 12]
```diff
@@ -16,7 +16,7 @@ type FilterButton = {
 const SearchBar = () => {
 	const { filter, updateFilter } = React.useContext(SearchContext);
 	const { stringQuery, updateStringQuery } = React.useContext(SearchContext);
-	const [barText, updateBarText] = React.useState(stringQuery);
+	const [ barText, updateBarText ] = React.useState(stringQuery);
```
[the remark this answers is not in the document]
Comment (AmauryDanisCousandier) [reply front/components/SearchBar.tsx:19]: that's strange I do not see that on my side...
image

File: front/components/SongRow.tsx (hunk 6 above)
Comment (AmauryDanisCousandier) [reply front/components/SongRow.tsx:14]: wdym ? SongRow.tsx is indeed the only file where this component has been created and exported

File: front/views/GenreDetailsView.tsx (hunk 10 above)
Comment (AmauryDanisCousandier) [reply front/views/GenreDetailsView.tsx:58]: I created a card grid custom component so w dont have to actually make a card grid component for each type but only the card component and then specify it to the CustomGrid component

**Review by Arthi-chaud — APPROVED** [review]

File: front/Theme.tsx (hunk 2 above)
Comment (Arthi-chaud) [reply front/Theme.tsx]: The documentation shows how we could do dynamic imports. This is not what we look for here.

**Review by Arthi-chaud — CHANGES_REQUESTED** [review]
Please, remove the `require` statement in Theme.tsx

**Review by Octopus773 — APPROVED** [review]
